pkg/v1/platform.go

```diff
@@ -28,11 +30,89 @@ type Platform struct {
 	Features     []string `json:"features,omitempty"`
 }
 
+func (p Platform) String() string {
+	if p.OS == "" {
+		return ""
+	}
+	var b strings.Builder
+	b.WriteString(p.OS)
+	if p.Architecture != "" {
+		b.WriteString("/")
+		b.WriteString(p.Architecture)
+	}
+	if p.Variant != "" {
+		b.WriteString("/")
+		b.WriteString(p.Variant)
+	}
+	if p.OSVersion != "" {
+		b.WriteString(":")
+		b.WriteString(p.OSVersion)
+	}
+	if len(p.OSFeatures) != 0 {
+		b.WriteString(" (osfeatures=")
+		b.WriteString(strings.Join(p.OSFeatures, ","))
+		b.WriteString(")")
+	}
+	if len(p.Features) != 0 {
+		b.WriteString(" (features=")
+		b.WriteString(strings.Join(p.Features, ","))
+		b.WriteString(")")
+	}
```

Comment:
I don't love this part but I love the rest.

Reply:
I thought you might feel that way. I can just drop that, but then roundtripping platforms that include these will become lossy. Almost no platforms in the wild will include these fields, so this codepath will basically never be exercised. I don't know whether that's an argument to allow this grossness to go in, or if it's an argument to just have lossy roundtrips in that case.

The rest of this will be really valuable, so I don't want to hang up too long on it. I'd rather just delete it. But I'll do whatever you think is right.

Reply:
I'm upset that my only objection is the syntax but because this is going to become a public interface, I think it actually  matters enough that I'd prefer lossy roundtrips (for now) :/